src/main.ts

```diff
@@ -85,56 +107,91 @@ async function main(): Promise<void> {
 
     await checkMarketplaceAddress(MarketplaceJSON.abi, marketplaceContract)
 
-    await watcher.on("productDeployed", async (...args: any[]): Promise<any> => {
-        const id = args[0]
-        const body = args[1]
+
+    await watcher.on("subscribed", async (args: {
+            blockNumber: number,
+            blockIndex: number,
+            product: string,
+            address: EthereumAddress,
+            endsAt: string,
+    }) => {
+        const {
+            blockNumber,
+            blockIndex,
+            product: productId,
+            address,
+            endsAt,
+        } = args
+        log.info("Watcher > subscribed event at block %s index %s: until %s", blockNumber, blockIndex, endsAt)
```

Comment:
These log messages are written from `main.ts` not `watchers.ts`. Looks like the prefix `Watcher > ...` is a bit misleading?